app/containers/EmojiPicker/index.tsx

```diff
@@ -1,204 +1,187 @@
-import React, { Component } from 'react';
-import { StyleProp, TextStyle, View } from 'react-native';
+import React, { useEffect, useState } from 'react';
+import { View } from 'react-native';
 import ScrollableTabView from 'react-native-scrollable-tab-view';
-import { dequal } from 'dequal';
-import { connect } from 'react-redux';
 import orderBy from 'lodash/orderBy';
 import { sanitizedRaw } from '@nozbe/watermelondb/RawRecord';
-import { ImageStyle } from 'react-native-fast-image';
 
 import TabBar from './TabBar';
 import EmojiCategory from './EmojiCategory';
+import Footer from './Footer';
 import styles from './styles';
 import categories from './categories';
 import database from '../../lib/database';
 import { emojisByCategory } from './emojis';
 import protectedFunction from '../../lib/methods/helpers/protectedFunction';
 import shortnameToUnicode from '../../lib/methods/helpers/shortnameToUnicode';
 import log from '../../lib/methods/helpers/log';
-import { themes } from '../../lib/constants';
-import { TSupportedThemes, withTheme } from '../../theme';
-import { IEmoji, TGetCustomEmoji, IApplicationState, ICustomEmojis, TFrequentlyUsedEmojiModel } from '../../definitions';
-
-interface IEmojiPickerProps {
-	isMessageContainsOnlyEmoji?: boolean;
-	getCustomEmoji?: TGetCustomEmoji;
-	baseUrl: string;
-	customEmojis: ICustomEmojis;
-	style?: StyleProp<ImageStyle>;
-	theme: TSupportedThemes;
-	onEmojiSelected: (emoji: string, shortname?: string) => void;
-	tabEmojiStyle?: StyleProp<TextStyle>;
-}
-
-interface IEmojiPickerState {
-	frequentlyUsed: (string | { content?: string; extension?: string; isCustom: boolean })[];
-	customEmojis: any;
-	show: boolean;
-	width: number | null;
-}
-
-class EmojiPicker extends Component<IEmojiPickerProps, IEmojiPickerState> {
-	constructor(props: IEmojiPickerProps) {
-		super(props);
-		const customEmojis = Object.keys(props.customEmojis)
-			.filter(item => item === props.customEmojis[item].name)
+import { useTheme } from '../../theme';
+import { IEmoji, ICustomEmojis, TFrequentlyUsedEmojiModel } from '../../definitions';
+import { useAppSelector } from '../../lib/hooks';
+import { IEmojiPickerProps, EventTypes } from './interfaces';
+
+const EmojiPicker = React.memo(
+	({
+		baseUrl,
+		onItemClicked,
+		tabEmojiStyle,
+		isEmojiKeyboard = false,
+		searching = false,
+		searchedEmojis = []
+	}: IEmojiPickerProps) => {
+		const [frequentlyUsed, setFrequentlyUsed] = useState<IEmoji[]>([]);
+		const [show, setShow] = useState(false);
+		const [width, setWidth] = useState(null);
+		const { colors } = useTheme();
+
+		const allCustomEmojis: ICustomEmojis = useAppSelector(state => state.customEmojis);
+		const customEmojis = Object.keys(allCustomEmojis)
```

Comment:
While as a Class Component, `constructor()` is triggered only once before the component actually mounting, as a Function Component everything is triggered on render all the time.
We should evaluate this and maybe turn it either into a custom hook or make use of `useMemo`